torch/_inductor/codegen/cpp_flex_attention_template.py

```diff
@@ -0,0 +1,1093 @@
+# mypy: allow-untyped-defs
+import contextlib
+import logging
+import re
+from typing import List, Optional
+from unittest.mock import patch
+
+import sympy
+
+import torch
+import torch.utils
+
+from .. import ir
+from ..ir import TensorBox
+from ..select_algorithm import DataProcessorTemplateWrapper
+from ..utils import parallel_num_threads
+from ..virtualized import V
+from .cpp_template import CppTemplate
+
+
+log = logging.getLogger(__name__)
+
+SOFTMAX_FUSIONS = r"""
+// 1) out = exp(a - val)
+// 2) val = sum(out)
+template <typename T1, typename T2>
+inline void {{kernel_name}}_exp_reduce_sum_fusion_kernel(
```

Comment:
Suppose we can reuse cpp codegen to generate these pointwise/reduction vectorized kernels? Can we add a TODO note here?

Reply:
yes, added TODO here.